docs/security/simple-ldap.md

```diff
@@ -45,54 +127,117 @@ plugin in the `CREATE USER` statement or the `ALTER USER` statement.:
+
+* authentication_ldap_simple_bind_root_dn
+
+* authentication_ldap_simple_bind_root_pwd
 
-    If the user is created with the “BY ‘cn,ou,dc,dc’” statement, the following variables are not used:
+* authentication_ldap_simple_user_search_attr
 
+* authentication_ldap_simple_group_search_attr
 
-        * authentication_ldap_simple_bind_base_dn
+If the user is created with “IDENTIFIED BY authentication_ldap_simple” the
+listed variables are used.
+
+### Specify authentication_ldap_sasl in the statement
+
+For simple LDAP authentication, you must specify the authentication_ldap_sasl
+plugin in the `CREATE USER` statement or the `ALTER USER` statement.:
+
+```text
+mysql> CREATE USER ... IDENTIFIED WITH authentication_ldap_sasl;
+```
+
+### Add the BY clause and specify the SASL authentication string
+
+```text
+mysql> CREATE USER ... IDENTIFIED WITH authentication_ldap_sasl BY 'cn=[user name],ou=[organization unit],dc=[domain component],dc=com'
+```
 
+If the user is created with the “BY ‘cn,ou,dc,dc’” statement, the following variables are not used:
 
-        * authentication_ldap_simple_bind_root_dn
+* authentication_ldap_sasl_bind_base_dn
 
+* authentication_ldap_sasl_bind_root_dn
 
-        * authentication_ldap_simple_bind_root_pwd
+* authentication_ldap_sasl_bind_root_pwd
 
+* authentication_ldap_sasl_user_search_attr
 
-        * authentication_ldap_simple_user_search_attr
+* authentication_ldap_sasl_group_search_attr
 
+If the user is created with “IDENTIFIED BY authentication_ldap_sasl” the
+listed variables are used.
 
-        * authentication_ldap_simple_group_search_attr
+## Examples
 
-    If the user is created with “IDENTIFIED BY authentication_ldap_simple” the
-    listed variables are used.
+The following sections are examples of using simple LDAP authentication and SASL-based LDAP authentication.
 
-If a MySQL user `rshimek` has the following entry in the LDAP directory:
+For the purposes of this example, we use the following OpenLDAP group:
 
 ```text
-uid=rshimek, ou=users, dc=hr, dc=com
+uid=ldapuser,ou=testusers,dc=percona,dc=com
 ```
 
-To create a MySQL account for `rshimek`, use the following statement:
+### Simple LDAP authentication
+
+The following example configures an LDAP user and connects to the database server.
+
+Create a database server account for `ldapuser` with the following statement:
+
+```sql
+mysql> CREATE USER 'ldapuser'@'localhost' IDENTIFIED WITH authentication_ldap_simple BY 'uid=ldapuser,ou=testusers,dc=percona,dc=com';
+```
+
+The authentication string does not include the LDAP password. This password must be provided by the client user when they connect.
+
+```sql
+mysql> mysql --user=ldapuser --password --enable-cleartext-plugin
+```
+
+The user enters the `ldapuser` password. The client sends the password as cleartext, which is necessary when using a server-side LDAP library without SASL. This requirement is a security problem in some environments. The following actions may minimize the risk:
```

Comment:
"This requirement is a security problem in some environments." - I'm not sure if I would add this, the SASL setup has its own security issues.

From a security perspective, ldap_simple with an SSL connection is more secure than the SASL plugin (with our without SSL), and the SASL plugin is more secure than ldap_simple without SSL.

Reply:
Removed.